### PR TITLE
feat: add frontend for locked workspaces

### PR DESCRIPTION
**Pull request by sreya**

### CODE REVIEW

File: site/src/components/Dashboard/DashboardProvider.tsx [hunk 1]
```diff
@@ -87,3 +87,13 @@ export const useDashboard = (): DashboardProviderValue => {
 
   return context
 }
+
+export const useIsWorkspaceActionsEnabled = (): boolean => {
```
Comment (BrunoQuaresma) [inline site/src/components/Dashboard/DashboardProvider.tsx:91]: Nice to encapsulate this logic making it easier to understand 💯

File: site/src/components/WorkspaceDeletion/ImpendingDeletionBadge.tsx [hunk 2]
```diff
@@ -1,31 +1,17 @@
...
-
-  if (
-    !displayImpendingDeletion(
-      workspace,
-      allowAdvancedScheduling,
-      allowWorkspaceActions,
-    )
-  ) {
+  const experimentEnabled = useIsWorkspaceActionsEnabled()
+  if (!workspace.locked_at || !experimentEnabled) {
```
Comment (BrunoQuaresma) [inline site/src/components/WorkspaceDeletion/ImpendingDeletionBadge.tsx:12]: Wondering if this check could not be done in the parent but no strong opinion.

File: site/src/pages/WorkspacesPage/WorkspacesPage.tsx [hunk 3]
```diff
@@ -21,6 +25,37 @@ const WorkspacesPage: FC = () => {
...
+            // TODO
+          })
+      }
+    } else {
+      // If the experiment isn't included then we'll pretend
+      // like locked workspaces don't exist.
+      setLockedWorkspaces([])
+    }
+  }, [experimentEnabled, data, filterProps.filter.query])
+
```
Comment (BrunoQuaresma) [inline site/src/pages/WorkspacesPage/WorkspacesPage.tsx:58]: This logic feels very similar to the `useWorkspacesToBeLocked`, why not use it here as well?

**Review by BrunoQuaresma — APPROVED** [review]
Just minor comments but overall, it looks good to go 👍

File: site/src/pages/WorkspacesPage/WorkspacesPage.tsx (hunk 3 above)
Comment (sreya) [reply site/src/pages/WorkspacesPage/WorkspacesPage.tsx:58]: The `useWorkspacesToBeLocked` takes a template which the workspaces page doesn't mandate. I could refactor it, but I'd rather merge this as-is and go back and clean up things as we move it out of experimental.

**Pull request merged**